src/interface/blas1/backend/amd_gpu.hpp

```diff
@@ -101,16 +134,43 @@ template <typename sb_handle_t, typename container_0_t, typename container_1_t,
 typename sb_handle_t::event_t _nrm2(
     sb_handle_t& sb_handle, index_t _N, container_0_t _vx, increment_t _incx,
     container_1_t _rs, const typename sb_handle_t::event_t& _dependencies) {
-  if (_N < (1 << 18)) {
-    constexpr index_t localSize = 1024;
-    const index_t number_WG = (_N + localSize - 1) / localSize;
-    return blas::internal::_nrm2_impl<static_cast<int>(localSize), 32>(
-        sb_handle, _N, _vx, _incx, _rs, number_WG, _dependencies);
+  /**
+   * Read comment in _asum above.
+   **/
+#ifdef SB_ENABLE_USM
+  bool usm_managed_mem{false};
+  if constexpr (std::is_pointer_v<decltype(_rs)>) {
+    usm_managed_mem =
+        sycl::usm::alloc::shared ==
+        sycl::get_pointer_type(_rs, sb_handle.get_queue().get_context());
+  }
+#else
+  constexpr bool usm_managed_mem{false};
+#endif
```

Comment:
Thank you @hjabird for your review.
I agree with you that adding a single function to `blas::helper` makes the code cleaner. However, I have one concern regarding the documentation. I don't think I can use the same I used for `_asum` because in that case, I specifically discuss the `reduction operator`, and outside of that context, it seems out of place to me. Instead, I prefer to create the function elsewhere but leave a comment above this code snippet and reference `nrm2` and `dot` to it.

To clarify, the function will be something like:

```c++
template <typename sb_handle_t, typename containerT>
inline bool is_malloc_shared(sb_handle_t &sb_handle,
                                   const containerT _rs) {
  if constexpr (std::is_pointer_v<decltype(_rs)>) {
    return sycl::usm::alloc::shared ==
           sycl::get_pointer_type(_rs, sb_handle.get_queue().get_context());
  }else {
    return false;
  }
}
```

As you can see, this function is generic, so adding a comment referring to atomics and reduction doesn't seem appropriate to me. What do you think?

Reply:
Nice! I'd be tempted to have two implementations of the function, one of which is constexpr for when `SB_ENABLE_USM` is undefined.

I agree that the comment that refers to AMD's atomics would be out of place. Perhaps we can put it with the if statement where the difference is? I'm not sure how to avoid the repetition of the comment in a good way. If we repeat it, maybe it can be shortened.

Reply:
I moved the function into `blas::helper` and update the comment in 5713422f46b3050266aaf6b705d42d47f72e7310